common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -243,6 +247,114 @@ public static void load(String name, ClassLoader loader) {
         }
     }
 
+    /**
+     * Loading the native library through the specified ClassLoader.
+     * @param loader - The specified ClassLoader where the native library will be loaded into
+     * @param name - The native library path or name
+     * @param absolute - Whether the native library will be loaded by path or not
+     */
+    private static void loadLibrary(final ClassLoader loader, final String name, final boolean absolute) {
+        try {
+            // Make sure the helper is belong to the target ClassLoader.
+            final Class<?> newHelper = tryToLoadClass(loader, NativeLibraryUtil.class);
+            loadLibraryByHelper(newHelper, name, absolute);
+        } catch (Exception e) { // Should by pass the UnsatisfiedLinkError here!
```

Comment:
The System.load() will throw a UnsatisfiedLinkError if the library is not found. If we cache Throwable here, the System.load() will be called twice. That is why I cache Exception here. WDYT?